docs/developer-docs/security/rust-canister-development-security-best-practices.md

```diff
@@ -10,20 +10,32 @@ This document contains canister development best practices for both Motoko and R
+1. **The dapp uses central, off-chain components:** The application makes use of centralized components that are e.g. running in the cloud. The owners of these cloud services have full control over the application and assets managed by it. Your application should likely be further decentralized by avoiding central components. But while you have them, [securely manage your keys in the cloud](https://cloudsecurityalliance.org/research/topics/cloud-key-management/). 
+2. **Centralized dapp control by the developer team:** Your project is not controlled by a DAO, for example because it is in an early development stage or does not (yet) hold significant funds. In that case, we recommend managing access to your canisters securely and ideally not letting individuals take full control of the application. To achieve that, consider requiring several individuals or parties to control canisters and to make security sensitive changes on the application level, such as modifying access controls, etc. A helpful tool to achieve this is the [threshold canister](https://github.com/dfinity/threshold). Manage key material using hardware security modules, such as e.g. [YubyHSM](https://www.yubico.com/ch/store/yubihsm-2-series/) and physically protect these. Some of these devices support threshold signature schemes, which can help to further secure the setup. 
+3. **Full decentralization using a DAO**: The dapp is controlled by a decentralized governance system such as the Internet Computer's [Service Nervous System (SNS)](../../developer-docs/integrations/sns), so that any security sensitive changes to the canisters are only executed if the SNS community approves them collectively through a proposal voting mechanism. If an SNS is used:
+   - Make sure voting power is distributed over many, independent entities such that there is not one single or a few entities that can decide by themselves how the DAO evolves, see [here](../../developer-docs/integrations/sns/tokenomics/tokenomics-intro#voting-power-and-decentralization).
+   - Ensure all components of the dapp are under SNS control, including the canisters serving the web frontends. 
+   - Consider the [SNS preparation checklist](../../developer-docs/integrations/sns/tokenomics/sns-checklist). Important points from a security perspective are e.g. [tokenomics](../../developer-docs/integrations/sns/tokenomics/sns-checklist#11-tokenomics-specification), [disclosing dependencies to off-chain components](../../developer-docs/integrations/sns/tokenomics/sns-checklist#13-disclosure-of-dependencies) and [performing security reviews](../../developer-docs/integrations/sns/tokenomics/sns-checklist#21-security-review).
+   - Use an SNS on the SNS subnet as this guarantees that the SNS is running an NNS-blessed version and maintained as part of ICP.
```

Comment:
```suggestion
   - Rather than [self-deploying the SNS code](https://internetcomputer.org/docs/current/developer-docs/integrations/sns/introduction/dao-alternatives#self-deploy-the-sns-code), use the official SNS on the SNS subnet, as this guarantees that the SNS is running an NNS-blessed version and maintained as part of ICP.
```

(I am not sure if we should only call it an "SNS" if it is on the SNS subnet, so I thought this might make it a bit clearer. Or do you think this goes to far? )

Reply:
After reading the last paragraph, I wonder to what extent we want to recommend this?
I think we should be open for alternative DAOs, but might also want to point out that the built-in SNS DAO comes with some security wins. I proposed to refrase it as "consider...". WDYT?